src/anomalib/metrics/per_image/pimo.py

```diff
@@ -0,0 +1,956 @@
+
+    anomaly_maps: list[Tensor]
+    masks: list[Tensor]
+
+    @property
+    def _is_empty(self) -> bool:
+        """Return True if the metric has not been updated yet."""
+        return len(self.anomaly_maps) == 0
+
+    @property
+    def num_images(self) -> int:
+        """Number of images."""
+        return sum([am.shape[0] for am in self.anomaly_maps])
+
+    @property
+    def image_classes(self) -> Tensor:
+        """Image classes (0: normal, 1: anomalous)."""
+        return _images_classes_from_masks(self.masks)
+
+    def __init__(
+        self,
+        num_threshs: int,
+        binclf_algorithm: BinclfAlgorithm | str = BinclfAlgorithm.NUMBA.value,
+    ) -> None:
+        """Per-Image Overlap (PIMO) curve.
+
+        Args:
+            num_threshs: number of thresholds used to compute the PIMO curve (K)
+            binclf_algorithm: algorithm to compute the binary classification curve (see `binclf_curve_numpy.Algorithm`)
+        """
+        super().__init__()
+
+        logger.warning(
+            f"Metric `{self.__class__.__name__}` will save all targets and predictions in buffer."
+            " For large datasets this may lead to large memory footprint.",
+        )
+
+        # the options below are, redundantly, validated here to avoid reaching
+        # an error later in the execution
+
+        _validate.is_num_threshs_gte2(num_threshs)
+        self.num_threshs = num_threshs
+
+        # validate binclf_algorithm and get string
+        self.binclf_algorithm = BinclfAlgorithm(binclf_algorithm).value
+
+        self.add_state("anomaly_maps", default=[], dist_reduce_fx="cat")
+        self.add_state("masks", default=[], dist_reduce_fx="cat")
+
+    def update(self, anomaly_maps: Tensor, masks: Tensor) -> None:
+        """Update lists of anomaly maps and masks.
+
+        Args:
+            anomaly_maps (Tensor): predictions of the model (ndim == 2, float)
+            masks (Tensor): ground truth masks (ndim == 2, binary)
+        """
+        _validate_is_anomaly_maps(anomaly_maps)
+        _validate_is_masks(masks)
+        _validate.is_same_shape(anomaly_maps, masks)
+        self.anomaly_maps.append(anomaly_maps)
+        self.masks.append(masks)
+
+    def compute(self) -> PIMOResult:
+        """Compute the PIMO curves.
+
+        Call the functional interface `pimo_curves()`, which is a wrapper around the numpy code.
+
+        Returns:
+            PIMOResult: PIMO curves dataclass object. See `PIMOResult` for details.
+        """
+        if self._is_empty:
+            msg = "No anomaly maps and masks have been added yet. Please call `update()` first."
+            raise RuntimeError(msg)
+        anomaly_maps = torch.concat(self.anomaly_maps, dim=0)
+        masks = torch.concat(self.masks, dim=0)
+        return pimo_curves(
+            anomaly_maps,
+            masks,
+            self.num_threshs,
+            binclf_algorithm=self.binclf_algorithm,
+        )
+
+
+class AUPIMO(PIMO):
```

Comment:
would it be an idea to have a dedicated module for this, even maybe sub-package if needed. 

Option I
```bash
metrics
  per_image
    aupimo.py
```

Option II
```bash
metrics
  per_image
    aupimo
      aupimo.py
      utils..
```